R/presto.R

```diff
@@ -1,15 +1,26 @@
 #' @export
-queryPresto <- function(host, port, username, password, schema, catalog, numOfRows = -1, query, ...){
+queryPresto <- function(host, port, username, password = "", schema, catalog, numOfRows = -1, query, ...){
+  if(!requireNamespace("httr")){stop("package httr must be installed.")}
+  if(!requireNamespace("stringr")){stop("package stringr must be installed.")}
+  if(!requireNamespace("RPresto")){stop("package RPresto must be installed.")}
+
 
   # read stored password
   pass = saveOrReadPassword("presto", username, password)
-  drv <- RPresto::Presto()
-  conn <- RPresto::dbConnect(drv, user = username,
-                                password = pass, host = host, port = port, schema = schema, catalog = catalog, session.timezone = Sys.timezone(location = TRUE))
-  # set envir = parent.frame() to get variables from users environment, not papckage environment
-  resultSet <- RPresto::dbSendQuery(conn, glue::glue_sql(query, .con = conn, .envir = parent.frame()))
-  df <- DBI::dbFetch(resultSet, n = numOfRows)
+  conn = getDBConnection("presto", host = host, port = port, databaseName = "", username = username, password = pass, catalog = catalog, schema = schema, dsn="", additionalParams = "",
```

Comment:
Please use arrow rather than equal for substitution. The previous line for pass too.

Reply:
Changed to use arrow.